### PR TITLE
Update contributors list, add codacy badge and clean code

### PR DESCRIPTION
**Pull request by rtfpessoa**
Since more people are contributing to the code Codacy will help
keep it consistent and easy to maintain, by suggesting improvements

### CODE REVIEW

File: src/printer-utils.js [hunk 1]
```diff
@@ -15,39 +15,41 @@
...
+    return 'd2h-' + hashCode(this.getDiffName(file)).toString().slice(-6);
   };
 
   PrinterUtils.prototype.getDiffName = function(file) {
     var oldFilename = file.oldName;
     var newFilename = file.newName;
 
-    if (oldFilename && newFilename
-      && oldFilename !== newFilename
-      && !isDeletedName(newFilename)) {
```
Comment (rtfpessoa) [inline src/printer-utils.js]: Codacy Issue found: ["isDeletedName" was used before it was defined](https://www.codacy.com/app/Codacy/diff2html/file/1624192717/issues/source?bid=3025341&fileBranchId=3033058#l40)

File: src/line-by-line-printer.js [hunk 2]
```diff
@@ -125,20 +128,21 @@
...
-        if (line.type == diffParser.LINE_TYPE.CONTEXT) {
-          lines += self.generateLineHtml(line.type, line.oldNumber, line.newNumber, escapedLine);
-        } else if (line.type == diffParser.LINE_TYPE.INSERTS && !oldLines.length) {
-          lines += self.generateLineHtml(line.type, line.oldNumber, line.newNumber, escapedLine);
-        } else if (line.type == diffParser.LINE_TYPE.DELETES) {
+
+        if (line.type === diffParser.LINE_TYPE.CONTEXT) {
+          lines += self._generateLineHtml(line.type, line.oldNumber, line.newNumber, escapedLine);
+        } else if (line.type === diffParser.LINE_TYPE.INSERTS && !oldLines.length) {
+          lines += self._generateLineHtml(line.type, line.oldNumber, line.newNumber, escapedLine);
```
Comment (rtfpessoa) [inline src/line-by-line-printer.js]: Codacy Issue found: [use `Boolean(oldLines.length)` instead.](https://www.codacy.com/app/Codacy/diff2html/file/1624192716/issues/source?bid=3025341&fileBranchId=3033058#l142)

File: src/side-by-side-printer.js [hunk 3]
```diff
@@ -130,36 +132,41 @@
...
+
+        if (line.type === diffParser.LINE_TYPE.CONTEXT) {
           fileHtml.left += generateSingleLineHtml(line.type, line.oldNumber, escapedLine, prefix);
           fileHtml.right += generateSingleLineHtml(line.type, line.newNumber, escapedLine, prefix);
-        } else if (line.type == diffParser.LINE_TYPE.INSERTS && !oldLines.length) {
+        } else if (line.type === diffParser.LINE_TYPE.INSERTS && !oldLines.length) {
           fileHtml.left += generateSingleLineHtml(diffParser.LINE_TYPE.CONTEXT, '', '', '');
           fileHtml.right += generateSingleLineHtml(line.type, line.newNumber, escapedLine, prefix);
-        } else if (line.type == diffParser.LINE_TYPE.DELETES) {
+        } else if (line.type === diffParser.LINE_TYPE.DELETES) {
```
Comment (rtfpessoa) [inline src/side-by-side-printer.js]: Codacy Issue found: [use `Boolean(oldLines.length)` instead.](https://www.codacy.com/app/Codacy/diff2html/file/1624192719/issues/source?bid=3025341&fileBranchId=3033058#l169)

File: src/side-by-side-printer.js [hunk 4]
```diff
@@ -130,36 +132,41 @@
...
-              var oldSlice = oldLines.slice(common),
-                  newSlice = newLines.slice(common);
-              tmpHtml = processLines(oldLines.slice(common), newLines.slice(common));
-              fileHtml.left += tmpHtml.left;
-              fileHtml.right += tmpHtml.right;
-            }
+          }
+
+          if (max > common) {
+            var oldSlice = oldLines.slice(common);
```
Comment (rtfpessoa) [inline src/side-by-side-printer.js]: Codacy Issue found: ["processLines" was used before it was defined](https://www.codacy.com/app/Codacy/diff2html/file/1624192719/issues/source?bid=3025341&fileBranchId=3033058#l141)

**Comment by pbu88** [comment]
Good catches all of them!! :+1:

**Comment by rtfpessoa** [comment]
@pbu88 thank Codacy and Intellij IDEA 15 :smiley:

**Comment by rtfpessoa** [comment]
@pbu88 when you added the config to the object, you implicitly re-assigned  the `self` object.

ESLint is complaining with this: http://eslint.org/docs/rules/no-native-reassign shouldn't we be doing something like `var self = this;`?

**Comment by pbu88** [comment]
Yeah totally, I saw that. We can either use `self` or `that`, I wasn't sure which one to pick up. Since self appears to be a native object maybe `that` would be more appropriate, and yes, I always forget adding `var ` in Javascript haha. I can do it right now, can you tell me which one do you prefer, if `that` or `self`?

**Comment by rtfpessoa** [comment]
I think since self is a native object we should use that to avoid confusion.

**Pull request merged**